### PR TITLE
Cert generator technique.

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**
Slight pivot from #32.

How to test:

- `npm i -g blackbaud/skyux-sdk-cli#cert-generator`
- `skyux certs trust`
- Create a new SPA but change the builder reference to `blackbaud/skyux-sdk-builder#cert-generator` (until https://github.com/blackbaud/skyux-sdk-builder/pull/168)
- Run `skyux serve`

I also took the opportunity to add OSX to the test suite since it was available in TravisCI and we support the platform.

### CODE REVIEW

File: lib/certs.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+async function untrustMac() {
+  await spawn(`sudo`, `security`, `delete-certificate`, `-c`, certUtils.getCertName());
+}
+
+async function trustWindows(argv) {
+  await spawn(`powershell`, `start-process cmd -verb runas -ArgumentList @("/c", "certutil", "-addstore", "-f", "root", '"${certUtils.getCertPath(argv)}"', "&", "PAUSE")`);
+}
+
+async function untrustWindows() {
+  await spawn(`powershell`, `start-process cmd -verb runas -ArgumentList @("/c", "certutil", "-delstore", "root", "${certUtils.getCertName()}", "&", "PAUSE")`);
```
Comment (Blackbaud-TerryHelems) [inline lib/certs.js]: Need to include the `'` so spaces in paths will get escaped.

File: lib/utils/cert-utils.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  logger.error(msg);
+  return false;
+}
+
+function validate(argv) {
+  const cert = getCertPath(argv);
+  const key = getKeyPath(argv);
+
+  if (!fs.existsSync(cert)) {
+    return error(`Error locating certificate: ${cert}`);
```
Comment (Blackbaud-SteveBrush) [inline lib/utils/cert-utils.js]: When I type `skyux certs untrust` and then `skyux certs validate`, I don't see this error message. Is that because it's seeing my "old" SKY UX certificate? Is there any way to clear the old one out too?

File: lib/utils/cert-utils.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  if (!fs.existsSync(key)) {
+    return error(`Error locating key: ${key}`);
+  }
+
+  // TODO: VALIDATE CERT + KEY using forge
```
Comment (Blackbaud-SteveBrush) [inline lib/utils/cert-utils.js]: Just confirming that we want to wait to implement this? If so, would it be worth creating an issue so we don't forget?

File: lib/utils/cert-utils.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+function validate(argv) {
+  const cert = getCertPath(argv);
+  const key = getKeyPath(argv);
+
+  if (!fs.existsSync(cert)) {
+    return error(`Error locating certificate: ${cert}`);
+  }
+
+  if (!fs.existsSync(key)) {
+    return error(`Error locating key: ${key}`);
```
Comment (Blackbaud-SteveBrush) [inline lib/utils/cert-utils.js]: If `skyux new` (or any other command) fails because the certificate is not found, would it be worth asking if the user ran `skyux certs trust`, just as a reminder? Something to the effect of...
```
`There was an error locating the certificate key \`${key}\`. Did you run \`skyux certs trust\``?
```

Same for the error above.

File: lib/utils/cert-utils.js (hunk 2 above)
Comment (Blackbaud-TerryHelems) [reply lib/utils/cert-utils.js]: It's not passing because it's finding your old cert, it's passing because those files still exist in `~\.skyux\certs`. I think the `TODO` a bit below is where that validation might occur.

Do we need that done now?

Comment (Blackbaud-BobbyEarl) [reply lib/utils/cert-utils.js]: Hmm... think `untrust` should also delete them?  I'm leaning towards yes.

Comment (Blackbaud-TerryHelems) [reply lib/utils/cert-utils.js]: That was my initial thought as well since `trust` always generates a new cert.

File: lib/utils/cert-utils.js (hunk 4 above)
Comment (Blackbaud-BobbyEarl) [reply lib/utils/cert-utils.js]: I call out a very similar message in two places.

1. In this PR at https://github.com/blackbaud/skyux-sdk-cli/pull/36/files#diff-168726dbe96b3ce427e7fedce31bb0bcR188-R191 Although I could reword it to mention running `skyux certs trust` as you suggested.

2. In the Builder PR at https://github.com/blackbaud/skyux-sdk-builder/pull/168/files#diff-78cf2d7696a73e189749f3a595673fafR7-R10

Do you think those are sufficient?

File: lib/utils/cert-utils.js (hunk 3 above)
Comment (Blackbaud-BobbyEarl) [reply lib/utils/cert-utils.js]: I think it would be nice to implement, but I was concerned with timing on finding the exact command to run.  Filling issue as https://github.com/blackbaud/skyux-sdk-cli/issues/37

File: lib/utils/cert-utils.js (hunk 4 above)
Comment (Blackbaud-SteveBrush) [reply lib/utils/cert-utils.js]: Yes, that will do! I think it would be helpful to explain the error and then what the user can do to solve it.

Comment (Blackbaud-BobbyEarl) [reply lib/utils/cert-utils.js]: Slightly updated the messaging for # 1.  Let me know if that doesn't resolve your suggestion.

File: lib/utils/cert-utils.js (hunk 2 above)
Comment (Blackbaud-BobbyEarl) [reply lib/utils/cert-utils.js]: Added a `remove` method that `untrust` calls.

Comment (Blackbaud-SteveBrush) [reply lib/utils/cert-utils.js]: Tested and it works!

**Comment by Blackbaud-BobbyEarl** [comment]
1. Added `certs` to list of commands.
2. I don't think so.  Users could still insecurely serve by clicking ignore in their browser.  We can revisit that decision if you like, but my vote is to warn as best we can but let it proceed.
